src/kakarot/instructions/system_operations.cairo

```diff
@@ -480,7 +478,7 @@ namespace CallHelper {
             read_only=read_only,
             is_create=FALSE,
         );
-        let sub_ctx = ExecutionContext.init(call_context);
+        let sub_ctx = ExecutionContext.init(call_context, 0);
```

Comment:
should we check above (in prepare_call?) that call_args.gas < ctx.gas_limit

Reply:
Indeed, even though not related to memory expansion, better do it rn. I’ll fix fist the tests and then just add this